tests/conftest.py

```diff
@@ -67,6 +76,67 @@ def fixture_login_page(selenium, server_url, org):
     return login_page.open()
 
 
+def create_user(server_url, session, user):
```

Comment:
I suggest to change this to be a closure, so that it's always running with the session of the root user without having to pass it in on every call.

If a user cannot be created, we should abort the test session by raising an exception.

We can make use of [formatted string literals](https://docs.python.org/3/whatsnew/3.6.html#pep-498-formatted-string-literals) since we're using Python 3.6! 🐍

```python
@pytest.fixture(name='root_user')
def fixture_root_user(variables, org):
    """Return the root user."""
    return User(**variables[org]['users']['root'])


@pytest.fixture(name='create_user', scope='session')
def fixture_create_user(server_url, root_user):
    """Return a function to create a user."""
    session = requests.Session()
    session.post(f"{server_url}/login", json={
        "email": root_user.email,
        "password": root_user.password,
    })

    def create_user(name=None, email=None, password=None):
        """Create user with the API and return a User instance."""
        response = session.post(f"{server_url}/api/users", json={
            "name": name,
            "email": email,
            "no_invite": True,
        }) 

        if response.status_code != 200:
            raise RuntimeError(f"unable to create user: {response.text")

        invite_link = response.json()["invite_link"]

        response = session.post(f"{server_url}{invite_link}", data=password)

        if response.status_code != 200:
            raise RuntimeError(f"unable to create user: {response.text")
        
        return User(name=name, email=email, password=password)

    return create_user


@pytest.fixture(name='user', scope='session')
def fixture_user(create_user, variables, org):
    """Return a registered user."""
    return create_user(**variables[org]['users']['ashley'])
```

Reply:
I'd rather leave the root login as its own thing as we will have to use that to do any kind of Admin interactions with the api. I could change it to a fixture and have one fixture just return the root user and another to login using it.

Reply:
Sounds good to me! 👍